### PR TITLE
Update generate_addon.sh

### PR DESCRIPTION
**Pull request by futuretea**
Signed-off-by: futuretea <Hang.Yu@suse.com>

**Related issues**
https://github.com/harvester/harvester/issues/2943

### CODE REVIEW

**Review by yaocw2020 — COMMENTED** [review]
Could you help modify the function `create_role` as a form like
``` shell 
cat <<EOF | kubectl apply -f -
......
EOF
```

**Comment by futuretea** [comment]
> Could you help modify the function `create_role` as a form like
> 
> ```shell
> cat <<EOF | kubectl apply -f -
> ......
> EOF
> ```

@yaocw2020 
Is there any particular reason?
It requires additional testing if it is modified, and it doesn't seem to be relevant to this issue

**Comment by futuretea** [comment]
> Could you help modify the function `create_role` as a form like
> 
> ```shell
> cat <<EOF | kubectl apply -f -
> ......
> EOF
> ```

OK

**Review by yaocw2020 — APPROVED** [review]
LGTM, thanks.

File: deploy/generate_addon.sh [hunk 1]
```diff
@@ -52,20 +52,52 @@ create_rolebinding() {
 }
 
 get_secret_name_from_service_account() {
-  echo -e "\\nGetting secret of service account ${SERVICE_ACCOUNT_NAME} on ${NAMESPACE}"
-  SECRET_NAME=$(kubectl get sa "${SERVICE_ACCOUNT_NAME}" --namespace="${NAMESPACE}" -o jsonpath="{.secrets[].name}")
+  SERVER_MINOR_VERSION=$(kubectl version -ojson | jq -r '.serverVersion.minor')
```
Comment (guangbochen) [inline deploy/generate_addon.sh:55]: a major + minor validation is preferred.

Comment (futuretea) [reply deploy/generate_addon.sh:55]: if major version >1, we still use the new logic 
do we need a check for major version == 0?

Comment (guangbochen) [reply deploy/generate_addon.sh:55]: `major version >= 1` is also fine, thanks.

Comment (futuretea) [reply deploy/generate_addon.sh]: updated

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Review by bk201 — APPROVED** [review]
LGTM

**Pull request merged**